From 285f317ff149ca414be132406726d9e9254b3366 Mon Sep 17 00:00:00 2001
From: Amitha Sanka<asanka@qti.qualcomm.com>
Date: Wed, 7 May 2025 11:35:11 -0700
Subject: [PATCH] phy: aquantia: Add eee support to AQR-github driver

Import fixes present in the no-ship driver into
AQR-github driver.

Signed-off-by: Amitha Sanka<asanka@qti.qualcomm.com>
---
 aquantia/aquantia_main.c | 30 +++++-------------------------
 1 file changed, 5 insertions(+), 25 deletions(-)

diff --git a/aquantia/aquantia_main.c b/aquantia/aquantia_main.c
index b0dd8f1..0f39d23 100644
--- a/aquantia/aquantia_main.c
+++ b/aquantia/aquantia_main.c
@@ -194,7 +194,7 @@
  * MDIO operation is finished
  */
 #define AQR107_OP_IN_PROG_SLEEP		1000
-#define AQR107_OP_IN_PROG_TIMEOUT	10000
+#define AQR107_OP_IN_PROG_TIMEOUT	50000
 
 #pragma pack(1)
 struct fw_header {
@@ -823,28 +823,6 @@ static int aqr113_set_wol(struct phy_device *phydev, struct ethtool_wolinfo *wol
 	return err;
 }
 
-static int aqr113_fix_provisioning(struct phy_device *phydev)
-{
-	int config_regs[] = {0x31B, 0x31C, 0x31D, 0x31E, 0x31F};
-	int i, val = 0;
-
-	for (i = 0; i < ARRAY_SIZE(config_regs); i++) {
-		val = phy_read_mmd(phydev, MDIO_MMD_VEND1, config_regs[i]);
-		printk(KERN_INFO"Set provisioning for reg:%x old value=%x\n", config_regs[i], val);
-#if IS_ENABLED(CONFIG_AQUANTIA_MACSEC)
-		/* Enabling MACSEC provisioning */
-		val |= BIT(9);
-#endif
-		/* Enabling EEE provisioning */
-		val |= BIT(11);
-
-		printk(KERN_INFO"Set provisioning for reg:%x new value=%x\n", config_regs[i], val);
-		phy_write_mmd(phydev, MDIO_MMD_VEND1, config_regs[i], val);
-	}
-
-	return 0;
-}
-
 /* Enable PHY load via MDIO */
 //#define MDIO_LOAD
 /* number of gangloaded ports to image load */
@@ -1016,8 +994,6 @@ static int aqr113_config_init(struct phy_device *phydev)
 
 	phydev->is_c45 = true;
 
-	aqr113_fix_provisioning(phydev);
-
 #if IS_ENABLED(CONFIG_AQUANTIA_MACSEC)
 	port->device = aqr_gen_4;
 	port->priv = phydev;
@@ -1106,6 +1082,10 @@ static int aqr107_probe(struct phy_device *phydev)
 	if (!phydev->priv)
 		return -ENOMEM;
 
+	/*To trigger renegotiation when link toggled*/
+	phydev->is_c45 = true;
+	phydev->c45_ids.devices_in_package = MDIO_DEVS_AN;
+
 	return aqr_hwmon_probe(phydev);
 }
 
-- 
2.34.1

